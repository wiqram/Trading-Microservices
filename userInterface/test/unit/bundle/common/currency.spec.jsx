import React from 'react';
import { create } from 'react-test-renderer';
import CurrencyConverter from '../../../../src/bundle/common/component/currency';

describe('CurrencyConverter Snapshot Test', () => {

    it('renders correctly with all props given -> £', () => {
        const amount = 1212.22;
        const type = 'CR';
        const currency = '£'
        const wrapper = create(<CurrencyConverter amount={amount} type={type} currency={currency}/>)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly with all props given -> $', () => {
        const amount = 1212.22;
        const type = 'DR';
        const currency = '$'

        const wrapper = create(<CurrencyConverter amount={amount} type={type} currency={currency}/>)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});