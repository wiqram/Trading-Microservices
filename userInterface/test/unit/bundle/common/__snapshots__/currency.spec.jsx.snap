// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyConverter Snapshot Test renders correctly with all props given -> $ 1`] = `
<div
  className="currency__TextStyle-sc-1uazvwe-0 gWoNzp"
  type="DR"
>
  $1,212.22
</div>
`;

exports[`CurrencyConverter Snapshot Test renders correctly with all props given -> Â£ 1`] = `
<div
  className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
  type="CR"
>
  $1,212.22
</div>
`;
